File: web/src/models/user.model.ts
export class UserModel {
  readonly #id: string;
  readonly #email: string;
  readonly #firstName: string;
  readonly #lastName: string;
  readonly #picture: string;

  constructor(
    id: string,
    email: string,
    firstName: string,
    lastName: string,
    picture: string
  ) {
    this.#id = id;
    this.#email = email;
    this.#firstName = firstName;
    this.#lastName = lastName;
    this.#picture = picture;
  }

  get id(): string {
    return this.#id;
  }

  get email(): string {
    return this.#email;
  }

  get firstName(): string {
    return this.#firstName;
  }

  get lastName(): string {
    return this.#lastName;
  }

  get picture(): string {
    return this.#picture;
  }
}

File: mobile/lib/models/user.model.dart
import 'package:equatable/equatable.dart';

class User extends Equatable {
  final String firstName;
  final String lastName;
  final String email;

  const User(
      {required this.firstName, required this.lastName, required this.email});

  @override
  List<Object?> get props => [firstName, lastName, email];

  static const empty = User(firstName: '', lastName: '', email: '');
}
